test/TokenFeeCollector.js

```diff
@@ -0,0 +1,181 @@
+        let [payeeAddress3, sharePercent3] = await tokenFeeCollector.viewPayee(WAVAX.address, 0)
+        expect(payeeAddress3).to.equal(account2.address)
+        expect(sharePercent3).to.equal(tenPercent)
+        await expect(tokenFeeCollector.viewPayee(WAVAX.address, 1)).to.be.reverted
+    })
+    
+    it('Others cant register token', async () => {
+        await expect(tokenFeeCollector.connect(account1).registerToken(WAVAX.address, account1.address, 1)).to.be.reverted;
+    })
+
+    it('Others cant add Payee', async () => {
+        const tenPercent = 100000000 * 10/100;
+        await tokenFeeCollector.registerToken(WAVAX.address, account1.address, tenPercent)
+        await expect(tokenFeeCollector.connect(account1).addPayee(WAVAX.address, account2.address, tenPercent+1)).to.be.reverted
+    })
+
+    it('Others cant edit Payee', async () => {
+        const tenPercent = 100000000 * 10/100;
+        await tokenFeeCollector.registerToken(WAVAX.address, account1.address, tenPercent)
+        await tokenFeeCollector.addPayee(WAVAX.address, account2.address, tenPercent+1)
+        await expect(tokenFeeCollector.connect(account1).editPayee(WAVAX.address, account1.address, tenPercent-1)).to.be.reverted
+    })
+
+    it('Others cant remove Payee', async () => {
+        const tenPercent = 100000000 * 10/100;
+        await tokenFeeCollector.registerToken(WAVAX.address, account1.address, tenPercent)
+        await tokenFeeCollector.addPayee(WAVAX.address, account2.address, tenPercent)
+        await expect(tokenFeeCollector.connect(account1).removePayee(WAVAX.address, account1.address)).to.be.reverted
+    })
+
+    it('Payee can view balance', async () => {
+        const maxBips = 100000000
+        const tenPercent = maxBips * 10/100
+        const eightyPercent = BigNumber.from(8).mul(tenPercent)
+        await tokenFeeCollector.registerToken(WAVAX.address, owner.address, eightyPercent)
+        await tokenFeeCollector.addPayee(WAVAX.address, account1.address, tenPercent)
+        await tokenFeeCollector.addPayee(WAVAX.address, account2.address, tenPercent)
+        const payment = 1000000000000
+        await WAVAX.transfer(tokenFeeCollector.address, payment)
+
+        let [sharePercent, balance] = await tokenFeeCollector.connect(owner).viewBalance(WAVAX.address)
+        expect(balance).to.equal(payment*eightyPercent/maxBips)
+        expect(sharePercent).to.equal(8*tenPercent)
+        let [sharePercent1, balance1] = await tokenFeeCollector.connect(account1).viewBalance(WAVAX.address)
+        expect(balance1).to.equal(payment*tenPercent/maxBips)
+        expect(sharePercent1).to.equal(tenPercent)
+        let [sharePercent2, balance2] = await tokenFeeCollector.connect(account2).viewBalance(WAVAX.address)
+        expect(balance2).to.equal(payment*tenPercent/maxBips)
+        expect(sharePercent2).to.equal(tenPercent)
+    })
+
+    it('Payee can collect fee', async () => {
```

Comment:
account2 never claims in this test

Reply:
When one payee of a given token claims, it claims for everybody, it's the simple way I found to give fair distribution